BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -108,6 +125,199 @@ public boolean onOptionsItemSelected(MenuItem item) {
+          writeBibFile("Download", shelfName, bibExportContent);
+        } else {
+          Toast.makeText(getContext(), R.string.storage_permission_denied,
+              Toast.LENGTH_SHORT).show();
+        }
+        break;
+
+      default:
+    }
+  }
+
+  private void createBibFile(String folderName, String fileName) {
+    try {
+      String rootPath = Environment.getExternalStorageDirectory() + "/" + folderName + "/";
+      File root = new File(rootPath);
+
+      if (!root.exists()) {
+        root.mkdirs();
+      }
+
+      File file = new File(rootPath + fileName + ".bib");
+      if (file.exists()) {
+        file.delete();
+      }
+      file.createNewFile();
+
+      FileOutputStream out = new FileOutputStream(file);
+      out.flush();
+      out.close();
+
+      Toast.makeText(getContext(), getString(R.string.exported_file_stored_in) + '\n'
+          + "/" + folderName + "/" + fileName + ".bib", Toast.LENGTH_LONG).show();
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void retrieveBibContent() {
+
+    DatabaseHelper dbHelper = new DatabaseHelper(getContext());
+    BookDao bd = new BookDao(dbHelper);
+    NoteDao nd = new NoteDao(dbHelper);
+
+    List<Long> bookIdsCurrShelf = bd.getAllBookIdsForShelf(shelfId);
+    List<Author> authorsCurrBook = new ArrayList<>();
+    List<Long> notesCurrBook = new ArrayList<>();
+    String allNotesCurrBook = "";
+    String authorNamesCurrBook = "";
+
+    //for each book in the current shelf
+    for (int i = 0; i < bookIdsCurrShelf.size(); i++) {
+      Book currBook;
+      Long currBookId = bookIdsCurrShelf.get(i);
+      authorsCurrBook = bd.getAllAuthorsForBook(currBookId);
+      notesCurrBook = nd.getAllNoteIdsForBook(currBookId);
+      currBook = bd.findById(currBookId);
+
+      /*
+      get the notes for the current book
+      and save the bib content in one string
+      */
+      for (int k = 0; k < notesCurrBook.size(); k++) {
+        String noteTextCurrBook = nd.findTextById(notesCurrBook.get(k));
+        allNotesCurrBook +=  "anote={" + noteTextCurrBook + "}," + '\n';
+      }
+
+      /*
+      get author's first and last name and include
+      the needed book data in a bib format
+      */
+      for (int u = 0; u < authorsCurrBook.size(); u++) {
+        authorNamesCurrBook = authorsCurrBook.get(u).getFirstName()
+            + " " + authorsCurrBook.get(u).getLastName();
+      }
+
+      bibExportContent = bibExportContent
+          + "@book{" + currBook.getTitle() + currBook.getPubYear() + "," + '\n'
+          + "isbn={" + currBook.getIsbn() + "}," + '\n'
+          + "author={" + authorNamesCurrBook + "}," + '\n'
+          + "title={" + currBook.getTitle() + "}," + '\n'
+          + "publisher={" + currBook.getPublisher() + "}," + '\n'
+          + "edition={" + currBook.getEdition() + "}," + '\n'
+          + allNotesCurrBook + "}," + '\n'
+          + "year=" + currBook.getPubYear() + '\n' + "}" + '\n';
```

Comment:
Maybe add another linebreak between the bibtex items, just for better readability in the exported file :)

```suggestion
          + "year=" + currBook.getPubYear() + '\n' + "}" + '\n\n';
```

Reply:
Your suggestion gives an error message, I will then write:

` + "year=" + currBook.getPubYear() + '\n' + "}" + '\n' + '\n';`

Reply:
Sounds good to me! Maybe '\n \n' works, too - have not checked that, however.